packages/repository-tests/src/crud/transactions.suite.ts

```diff
@@ -0,0 +1,146 @@
+export function transactionSuite(
+  dataSourceOptions: DataSourceOptions,
+  repositoryClass: TransactionalRepositoryCtor,
+  connectorFeatures: CrudFeatures,
+) {
+  skipIf<[(this: Suite) => void], void>(
+    !connectorFeatures.supportsTransactions,
+    describe,
+    `transactions`,
+    () => {
+      @model()
+      class Product extends Entity {
+        @property({
+          type: connectorFeatures.idType,
+          id: true,
+          generated: true,
+          description: 'The unique identifier for a product',
+        })
+        id: number | string;
+
+        @property({type: 'string', required: true})
+        name: string;
+
+        constructor(data?: Partial<Product>) {
+          super(data);
+        }
+      }
+
+      describe('create-retrieve with transactions', () => {
+        let repo: TransactionalEntityRepository<
+          Product,
+          typeof Product.prototype.id
+        >;
+        let tx: Transaction | undefined;
+        before(
+          withCrudCtx(async function setupRepository(ctx: CrudTestContext) {
+            repo = new repositoryClass(Product, ctx.dataSource);
+            await ctx.dataSource.automigrate(Product.name);
+          }),
+        );
+        beforeEach(() => {
+          tx = undefined;
+        });
+        afterEach(async () => {
+          if (tx !== undefined && tx.connection) {
+            await tx.rollback();
+          }
+        });
+
+        it('retrieves model instance once transaction is committed', async () => {
+          tx = await repo.beginTransaction({
+            isolationLevel: IsolationLevel.READ_COMMITTED,
+          });
+          const created = await repo.create(
+            {name: 'Pencil'},
+            {transaction: tx},
+          );
+          expect(created.toObject()).to.have.properties('id', 'name');
+          expect(created.id).to.be.ok();
+
+          await tx.commit();
+          const foundOutsideTransaction = await repo.findById(created.id, {});
+          expect(toJSON(created)).to.deepEqual(toJSON(foundOutsideTransaction));
+        });
+
+        it('can rollback transaction', async () => {
+          tx = await repo.beginTransaction({
+            isolationLevel: IsolationLevel.READ_COMMITTED,
+          });
+
+          const created = await repo.create(
+            {name: 'Pencil'},
+            {transaction: tx},
+          );
+          expect(created.toObject()).to.have.properties('id', 'name');
+          expect(created.id).to.be.ok();
+
+          const foundInsideTransaction = await repo.findById(
+            created.id,
+            {},
+            {
+              transaction: tx,
+            },
+          );
+          expect(toJSON(created)).to.deepEqual(toJSON(foundInsideTransaction));
+          await tx.rollback();
+          await expect(repo.findById(created.id, {})).to.be.rejectedWith({
+            code: 'ENTITY_NOT_FOUND',
+            message: /Entity not found/,
```

Comment:
It's enough to assert on the `code`. The message can change in the future, we don't want this test to fail if that happens. Please remove this line.